src/Graphics/OpenGLContext/opengl_BufferedDrawer.cpp

```diff
@@ -0,0 +1,218 @@
+#include <Graphics/Context.h>
+#include <Config.h>
+#include "GLFunctions.h"
+#include "opengl_Attributes.h"
+#include "opengl_CachedFunctions.h"
+#include "opengl_BufferedDrawer.h"
+
+using namespace graphics;
+using namespace opengl;
+
+BufferedDrawer::BufferedDrawer(const GLInfo & _glinfo, CachedVertexAttribArray * _cachedAttribArray, CachedBindBuffer * _bindBuffer)
+: m_glInfo(_glinfo)
+, m_cachedAttribArray(_cachedAttribArray)
+, m_bindBuffer(_bindBuffer)
+{
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::position, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::color, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::texcoord, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::numlights, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::modify, false);
+
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::position, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::color, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::texcoord0, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::texcoord1, false);
+
+	int fake;
+	m_attribsData.fill(&fake);
+	m_bufMaxSize = 4194304;
```

Comment:
I had circa 24mb for all buffers and textures (including HD ones) on my Voodoo5. Now I have gigs of video memory, but still can't used to it. 4MB just to be sure that it is enough! Such buffer can hold circa 95000 vertex elements btw. That is, at least for 100 frames. Of course, it is overkill.
I left that limit in my implementation for now, but it is matter for future optimizations.